frame/treasury/src/lib.rs

```diff
@@ -492,6 +505,8 @@ decl_module! {
 			let tips = vec![(tipper, tip_value)];
 			let tip = OpenTip { reason: reason_hash, who, finder: None, closes: None, tips };
 			Tips::<T>::insert(&hash, tip);
+
+			Ok(Some(MAX_TIPPERS_COUNT.saturating_sub(T::Tippers::count() as u64) * 481_000).into())
```

Comment:
I wonder if there is a clean way to turn this raw number into a variable so that updates are less prone to a mistake here.

Reply:
we can still introduce a new constant before decl_module, it is the only solution I see

Reply:
probably not worth it tbh, makes the code even more fragmented. Please add a comment, I think that will be enough

Reply:
I agree, but I reimplemented without refund now.

Reply:
This looks like you are calculating the "weight not spent" (because you are subtracting tippers from max_tippers) but I think you are meant to return actual weight used, so it would need to be something like
```
Ok(Some(107_700_000 + 4_000 * reason.len() as u64 + 481_000 * T::Tippers::count() + T::DbWeight::get().reads_writes(2, 2)))
```
no?

Reply:
It would be really cool if we had access to the weight calculated in the macro here so we could just subtract similar to how you did.

Reply:
you are right, I thought it was the weight to refund but no it is the actual weight used so your formula is correct.